main.tf

```diff
@@ -109,26 +123,17 @@ resource "ibm_is_vpc_dns_resolution_binding" "vpc_dns_resolution_binding_id" {
 
 resource "ibm_is_vpc_dns_resolution_binding" "vpc_dns_resolution_binding_crn" {
   count  = (var.enable_hub == false && var.enable_hub_vpc_crn) ? 1 : 0
-  name   = "${var.prefix}-dns-binding"
+  name   = var.prefix != null ? "${var.prefix}-dns-binding" : var.dns_binding_name
   vpc_id = local.vpc_id # Source VPC
   vpc {
     crn = var.hub_vpc_crn # Target VPC CRN
   }
 }
 
-data "ibm_is_vpc" "vpc" {
-  count      = var.create_vpc == false ? 1 : 0
-  identifier = var.existing_vpc_id
-}
-
-locals {
-  vpc_id = var.create_vpc ? ibm_is_vpc.vpc[0].id : var.existing_vpc_id
-}
-
 # Configure custom resolver on the hub vpc
 resource "ibm_resource_instance" "dns_instance_hub" {
   count             = var.enable_hub && !var.skip_custom_resolver_hub_creation && !var.use_existing_dns_instance ? 1 : 0
-  name              = "${var.prefix}-dns-instance"
+  name              = var.prefix != null ? "${var.prefix}-dns-instance" : var.dns_instance_name
```

Comment:
Regardless of the value of `var.prefix`, if the user passes a value for `var.dns_instance_name` the code should use that. Otherwise use "${var.prefix}-${var.name}-dns-instance" unless prefix is null, in which case use "${var.name}-dns-instance"